cmd/api-endpoint/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/edulinq/autograder/internal/api"
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/cmd"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/exit"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+var args struct {
+	config.ConfigArgs
+	cmd.CommonOptions
+
+	Endpoint   string   `help:"Endpoint of the desired API." arg:""`
+	Parameters []string `help:"Parameter(s) for the endpoint" arg:"" optional:""`
+	Table      bool     `help:"Output data as a TSV." default:"false"`
+}
+
+func main() {
+	kong.Parse(&args,
+		kong.Description("Execute an API request to the specified endpoint."),
+	)
+
+	err := config.HandleConfigArgs(args.ConfigArgs)
+	if err != nil {
+		log.Fatal("Failed to load config options.", err)
+	}
+
+	var description core.EndpointDescription
+
+	describe := api.Describe(*api.GetRoutes())
+	for endpoint, requestResponse := range describe.Endpoints {
+		if endpoint == args.Endpoint {
+			description = requestResponse
+			break
+		}
+	}
+
+	if description == (core.EndpointDescription{}) {
+		log.Error("Failed to find the endpoint.", log.NewAttr("endpoint", args.Endpoint))
+
+		if !config.UNIT_TESTING_MODE.Get() {
+			fmt.Print(util.MustToJSONIndent(api.Describe(*api.GetRoutes())))
+		}
```

Comment:
I don't see a reason for this distinction.
And remember, I mentioned before how we don't want to use this flippantly.
We only want to use this if we have to, we want our code to generally run the same in tests and production.
(The closer they are the less likely we are to have strange issues we only see in production).

I also question if you need this at all.
You can just point them to the API resource file in the help output.
(You could also enumerate them all in the help output, but I don't really see a need.)